libyul/backends/evm/EVMObjectCompiler.h

```diff
@@ -34,7 +34,7 @@ class EVMObjectCompiler
 {
 public:
 	static void compile(
-		Object& _object,
+		Object const& _object,
```

Comment:
Not sure why this wasn't `const` already. It should have been because `EVMObjectCompiler` does not modify the object. I'm changing it here because I wanted to be sure that `optimize()` is the only place that actually modifies the object and requires reparsing.

Reply:
True. But at least your immutable AST PR does address it, at least for the AST which is the most relevant part of the object. Perhaps we should just make `Object` a proper class and hide other members behind const accessors too.